fearless/Modules/NewsVersion2/EducationStoriesProgressView.swift

```diff
@@ -0,0 +1,158 @@
+import UIKit
+
+protocol EducationStoriesProgressViewDelegate: AnyObject {
+    func storiesProgressViewChanged(index: Int)
+    func storiesProgressViewFinished()
+}
+
+final class EducationStoriesProgressView: UIStackView {
+    weak var delegate: EducationStoriesProgressViewDelegate?
+
+    // MARK: - Constants
+
+    private enum Constants {
+        static let duration: TimeInterval = 10.0
+        static let animationDelayLong: TimeInterval = 0.2
+        static let animationDelayShort: TimeInterval = 0.1
+    }
+
+    // MARK: - Private properties
+
+    private var currentAnimationIndex = 0
+    private lazy var animator = UIViewPropertyAnimator()
+    private var isValid: Bool {
+        currentAnimationIndex < arrangedSubviews.count
+    }
+
+    // MARK: - Constructor
+
+    init() {
+        super.init(frame: CGRect.zero)
+        setup()
+    }
+
+    @available(*, unavailable)
+    required init(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public func
+
+    func configure(segmentsCount: Int) {
+        for _ in 0 ..< segmentsCount {
+            addArrangedSubview(createProgressView())
+        }
+        startAnimation()
+    }
+
+    func startAnimation() {
+        next()
+    }
+
+    func pauseAnimation() {
+        animator.pauseAnimation()
+    }
+
+    func continueAnimation() {
+        animator.continueAnimation(withTimingParameters: nil, durationFactor: 0)
+    }
+
+    func skip() {
+        if currentAnimationIndex <= arrangedSubviews.count, currentAnimationIndex > 0 {
+            if let currentSegment = arrangedSubviews[currentAnimationIndex - 1] as? UIProgressView {
+                stopAnimation(for: currentSegment, progress: 1)
+                next()
+            }
+        } else {
+            next()
+        }
+    }
+
+    func back() {
+        if currentAnimationIndex > 0 {
+            currentAnimationIndex -= 1
+
+            if let currentSegment = arrangedSubviews[currentAnimationIndex] as? UIProgressView {
+                stopAnimation(for: currentSegment, progress: 0)
+            }
+        }
```

Comment:
Need call twice because need stoped animation for current segment and stop animation for previous segment and start it first. Transferd this code in method